Automattic/wp-calypso
# Domains: Prevent Cancel and DNS section `render` for pending Domain Transfer

## Proposed Changes
We need to prevent some domain management sections from being displayed for pending domain transfers. This PR prevents the `render()` of the cancel card and the DNS section on the management page:

![image](https://github.com/Automattic/wp-calypso/assets/18705930/d7b2701c-d7fa-46da-a22d-faf4346735cf)


## Testing Instructions
- Follow the instructions described here: 30e76-pb/#plain;
- Go to the domain transfer management page;
- Make sure the cards are not displayed.

## Pre-merge Checklist
